x-pack/plugins/alerting/server/plugin.ts

```diff
@@ -440,11 +442,13 @@ export class AlertingPlugin {
 
   private getServicesFactory(
     savedObjects: SavedObjectsServiceStart,
-    elasticsearch: ElasticsearchServiceStart
+    elasticsearch: ElasticsearchServiceStart,
+    data: DataPluginStart
   ): (request: KibanaRequest) => Services {
     return (request) => ({
       savedObjectsClient: this.getScopedClientWithAlertSavedObjectType(savedObjects, request),
       scopedClusterClient: elasticsearch.client.asScoped(request),
+      searchSourceClient: data.search.searchSource.asScoped(request),
```

Comment:
Note for the real implementation: we won't pass in the data plugin or something via alerting, but alerting will only pass us in the faked KiabanRequest and we need to scope the search source in our rule implementation.